anchore_engine/services/catalog/catalog_impl.py

```diff
@@ -1826,6 +1873,30 @@ def perform_policy_evaluation(
     return curr_evaluation_record, curr_evaluation_result
 
 
+ImageKey = collections.namedtuple("ImageKey", ["tag", "digest"])
```

Comment:
this bit is new, and meant to support other features who _have_ an image tag and digest, and want it to be easy to add an image for analysis

Reply:
Note: this flow will NOT create any subscriptions for the image, that bit happens currently in the external API service

Reply:
any reason why we don't use dataclasses instead of namedtuples? we're using 3.8 now so compatibility isn't an issue anymore.

Reply:
https://medium.com/@jacktator/dataclass-vs-namedtuple-vs-object-for-performance-optimization-in-python-691e234253b9

Reply:
I didn't even know about data classes! I used this just because I was more familiar. There are some python 3.6 relics left in engine (around integration tests, I believe), so I am reticent to adding this now, but super glad you shared this so I know about it for the future!